superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -194,28 +194,44 @@ class AnnotationLayer extends React.PureComponent {
       hideLine,
       // refData
       isNew: !name,
-      isLoadingOptions: true,
-      valueOptions: [],
+      valueOptions: {},
```

Comment:
`valueOptions` is not being set anymore. You can remove it.

Reply:
Getting errors when this isn't initialized in state upon editing an older Annotation

Reply:
Forgot to remove `prevState` param from `setState`, should be good now